server/src/main/scala/ch.epfl.scala.index.server/routes/FrontPage.scala

```diff
@@ -34,12 +37,27 @@ class FrontPage(dataRepository: ESRepo, session: GithubUserSession)(implicit
     val contributingProjectsF = getContributingProjects()
 
     for {
-      topics <- topicsF
-      targetTypes <- targetTypesF
-      scalaVersions <- scalaVersionsF
-      scalaJsVersions <- scalaJsVersionsF
-      scalaNativeVersions <- scalaNativeVersionsF
-      sbtVersions <- sbtVersionsF
+      topics <- topicsF.map(FrontPage.getTopTopics(_, 50))
+      allPlatforms <- allPlatformsF
+      platformTypeWithCount = FrontPage.getPlatformTypeWithCount(allPlatforms)
+      scalaFamilyWithCount = FrontPage.getScalaLanguageVersionWithCount(
+        allPlatforms
+      )
+      scalaJsVersions = FrontPage
+        .getPlatformWithCount(allPlatforms) { case p: Platform.ScalaJs =>
+          p.scalaJsV
+        }
+        .sorted
```

Comment:
This `sorted` is repeated 5 times. It could be put in `getPlatformWithCount` instead.